src/Stratis.Bitcoin.Features.Miner/Staking/PosMinting.cs

```diff
@@ -647,6 +647,8 @@ public async Task<bool> CreateCoinstakeAsync(List<UtxoStakeDescription> utxoStak
             if (!stakingUtxoDescriptions.Any())
             {
                 this.rpcGetStakingInfoModel.Staking = false;
+                this.rpcGetStakingInfoModel.Weight = balance;
+                this.rpcGetStakingInfoModel.ExpectedTime = balance != 0 ? StakeValidator.TargetSpacingSeconds * this.networkWeight / this.rpcGetStakingInfoModel.Weight : 0;
```

Comment:
maybe here instead of dividing by this.rpcGetStakingInfoModel.Weight, I would divide by balance, just makes it a bit easier to follow (or compare this.rpcGetStakingInfoModel.Weight to 0 instead of balance). No big deal for me :)

Reply:
yeah, make sense.

Reply:
Basically something like we do online 361

Reply:
And also it should probably be done on line 638

Reply:
@Aprogiena won't it confuse? cause you didn't actually stop staking as a user, its just node waiting till your UTXOs get some depth because you just spent some coins.
And you were talking about "stop" staking. Yes, when staking is stopped, values will be reset to 0.

Reply:
The original argument in the issue was that it is confusing when it is not reset to 0 and the staking is not in progress. Basically we say that you will mine a block in about one hour (for example) in that information but at the same time you are not staking at all, so obviously you will not mine a block about one hour.

See the discussion on the issue. I argued that all values should be ignored if staking is not in progress and therefore does not matter which values are there. Dan was of an opinion that it doesn't matter and therefore it should be reset to 0, which I don't have a problem with, so this is what I was asking for here.

Reply:
In any case, whichever option we choose, it should be implemented in all three places, not only at one place

Reply:
Okey mate, it's easy to me, I will do reset in 3 places